modules/gapi/src/backends/ocl/goclcore.cpp

```diff
@@ -523,6 +541,47 @@ GAPI_OCL_KERNEL(GOCLTranspose, cv::gapi::core::GTranspose)
     }
 };
 
+GAPI_OCL_KERNEL(GOCLBGR, cv::gapi::streaming::GBGR)
+{
+    static void run(const cv::MediaFrame& in, cv::UMat& out)
+    {
+        (void)in; (void)out;
+#ifdef HAVE_DIRECTX
+#ifdef HAVE_D3D11
+#ifdef HAVE_ONEVPL
+        auto d = in.desc();
+        GAPI_Assert(d.fmt == cv::MediaFormat::NV12);
+
+        // FIXME: consider a better solution.
+        // Current approach cannot be easily extended for other adapters (getHandle).
+        auto params = in.get<cv::gapi::wip::onevpl::VPLMediaFrameDX11Adapter>()->getHandle();
```

Comment:
This is the first issue that needs to be addressed. For now it's working, but if we are going to extend that with something besides DX11 there might be some problems. But it can be reworked along with the future extension.

Reply:
we discussed `visitor/acceptor` approach with Maxim Paschenkov

Reply:
I'd better check result of `in.get<>()` right before dereferencing a pointer

Reply:
just get `cv::Mat` and call `getUMat()`?

Reply:
Can this discussion be closed?

Reply:
I think yes - we should close it.
But please pay the attention:
```
else {
    // If unknown adapter just sync to host
    auto view = in.access(READ); 
    // transfer to ocl;
 --->view data will be invalid when goes out of scope
---> Make sure that UMat copies data
}
```
I don't close this thread otherwise my last comment becomes hidden. Please feel free to close it by yourself